### ISSUE TITLE
Support Cluster Upgrade through automation

### ISSUE THREAD

**Issue by livespotty** [oracle-terraform-modules/terraform-oci-oke#178]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

    Upgrade Master Nodes on an existing terraform deployed oke cluster
    Upgrade Worker Nodes on an existing terraform deployed oke cluster

### Potential Terraform Configuration

```hcl
# Copy-paste any Terraform configurations for how the requested feature may be used. 
```

### References

https://docs.cloud.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengupgradingk8smasternode.htm
https://docs.cloud.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengupgradingk8sworkernode.htm

**Comment by hyder** [comment]
depends on #179

**Comment by hyder** [comment]
We'll support out of place upgrades in the beginning and we'll add in-place upgrades subsequently. https://docs.cloud.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengupgradingk8sworkernode.htm

**Comment by hyder** [comment]
Closed in #184
